### PR TITLE
[Playlist] Show 'New playlist' and 'Back' option as clickable

### PR DESCRIPTION
**Pull request by jagadeshjai**
- Create and use customized label button to show 'New playlist' and 'Back' option is a clickable button in `Move Media` dialog

**Outcome:**

[brave_playlist_button_fix_2.webm](https://github.com/user-attachments/assets/9f995c2a-ff5b-4cc2-b690-0d32887ed148)

Resolves https://github.com/brave/brave-browser/issues/40243

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by jagadeshjai** [comment]
@aguscruiz Could you share me the design for dialog?

**Comment by aguscruiz** [comment]
Hi @jagadeshjai, here you go, let me know if you need anything else. It's pretty close to what you have



https://www.figma.com/design/OW5NrP9ucBHl6tGQlt9zTE/Jagadesh---External-contributor?node-id=17-2499

**Comment by jagadeshjai** [comment]
> Hi @jagadeshjai, here you go, let me know if you need anything else. It's pretty close to what you have
> 
> https://www.figma.com/design/OW5NrP9ucBHl6tGQlt9zTE/Jagadesh---External-contributor?node-id=17-2499

Thanks @aguscruiz ! I'll check.

**Comment by jagadeshjai** [comment]
@aguscruiz 
I'm curious why we aren't changing the cursor to a hand pointer when hovering over buttons in dialogs and other areas. Is this an intended behavior? 

Here's an example from the Playlist section:

[Screencast from 10-08-24 10:22:32 AM IST.webm](https://github.com/user-attachments/assets/0edbb18b-91f3-4fc3-b1ea-621005fa4385)


CC: @sangwoo108

File: browser/ui/views/playlist/playlist_action_dialogs.cc [hunk 1]
```diff
@@ -251,6 +253,32 @@ void ClosePanel(content::WebContents* contents) {
...
+                      const std::u16string& text = std::u16string())
+      : LabelButton(std::move(callback), text) {
+    SetEnabledTextColorIds(kColorBravePlaylistTextInteractive);
+    label()->SetFontList(label()->font_list().Derive(
+        0, gfx::Font::FontStyle::NORMAL, gfx::Font::Weight::NORMAL));
+  }
+
+  ~PlaylistLabelButton() override = default;
+
+  ui::Cursor GetCursor(const ui::MouseEvent& event) override {
```
Comment (jagadeshjai) [inline browser/ui/views/playlist/playlist_action_dialogs.cc:270]: Note: Intentionally changing the cursor to a hand pointer, even though it is not done for other buttons in the dialog.

Need some clarification for this ( https://github.com/brave/brave-core/pull/25003#issuecomment-2279263095 )

**Comment by aguscruiz** [comment]
> I'm curious why we aren't changing the cursor to a hand pointer when hovering over buttons in dialogs and other areas. Is this an intended behavior?

I think this is inherited Chromium behavior. 
@simonhong @fallaciousreasoning correct me if I'm wrong here.

**Comment by sangwoo108** [comment]
> > I'm curious why we aren't changing the cursor to a hand pointer when hovering over buttons in dialogs and other areas. Is this an intended behavior?
> 
> I think this is inherited Chromium behavior. @simonhong @fallaciousreasoning correct me if I'm wrong here.

Yes, you're right. Chromium doesn't change cursor to pointer even for the clickable button. I have no preference for that but native desktop applications seem to rarely do that, unlike web applications.

File: browser/ui/views/playlist/playlist_action_dialogs.cc (hunk 1 above)
Comment (sangwoo108) [reply browser/ui/views/playlist/playlist_action_dialogs.cc:270]: For this button, I agree that it'd be better to change the cursor to hand, as the button seems to have no other interaction.

File: browser/ui/views/playlist/playlist_action_dialogs.cc [hunk 2]
```diff
@@ -251,6 +253,32 @@ void ClosePanel(content::WebContents* contents) {
...
+namespace {
+class PlaylistLabelButton : public views::LabelButton {
+  METADATA_HEADER(PlaylistLabelButton, views::LabelButton)
+ public:
+  PlaylistLabelButton(views::Button::PressedCallback callback = {},
+                      const std::u16string& text = std::u16string())
+      : LabelButton(std::move(callback), text) {
+    SetEnabledTextColorIds(kColorBravePlaylistTextInteractive);
+    label()->SetFontList(label()->font_list().Derive(
+        0, gfx::Font::FontStyle::NORMAL, gfx::Font::Weight::NORMAL));
```
Comment (sangwoo108) [inline browser/ui/views/playlist/playlist_action_dialogs.cc:265]: I think `SEMIBOLD` would have value of 600, which matches our design guide.

```suggestion
        0, gfx::Font::FontStyle::NORMAL, gfx::Font::Weight::SEMIBOLD));
```

Comment (jagadeshjai) [reply browser/ui/views/playlist/playlist_action_dialogs.cc]: Done! Also changed the font size to 13px as per you suggestion from [here](https://github.com/brave/brave-browser/issues/40243#issuecomment-2276924862).

**Comment by aguscruiz** [comment]
@jagadeshjai I also approved your dev mode request for Figma :)

**Comment by sangwoo108** [comment]
https://github.com/brave/brave-core/pull/25079 CI passed. Let me ping someone who can merge this

**Comment by fallaciousreasoning** [comment]
@sangwoo108 if you mark the other PR as ready for review does this one pass?

**Comment by sangwoo108** [comment]
> @sangwoo108 if you mark the other PR as ready for review does this one pass?

Unfortunately, it doesn't seem to work 😞

**Comment by jagadeshjai** [comment]
Thanks @bsclifton 😄

**Pull request merged**